src/PluginSuppression.php

```diff
@@ -44,15 +46,42 @@ public function __construct( PluginRegistry $plugin_registry ) {
 
 	/**
 	 * Register the service with the system.
-	 *
-	 * @return void
 	 */
 	public function register() {
-		$priority = defined( 'PHP_INT_MIN' ) ? PHP_INT_MIN : ~PHP_INT_MAX; // phpcs:ignore PHPCompatibility.Constants.NewConstants.php_int_minFound
-		add_action( 'wp', [ $this, 'suppress_plugins' ], $priority );
 		add_filter( 'amp_default_options', [ $this, 'filter_default_options' ] );
 		add_action( 'amp_options_menu_items', [ $this, 'add_settings_field' ], 11 );
 		add_filter( 'amp_options_updating', [ $this, 'sanitize_options' ], 10, 2 );
+		$priority = defined( 'PHP_INT_MIN' ) ? PHP_INT_MIN : ~PHP_INT_MAX; // phpcs:ignore PHPCompatibility.Constants.NewConstants.php_int_minFound
+
+		// When a Reader theme is selected and and an AMP request is being made, start suppressing as early as possible
+		// This can be done because we know it is an AMP page due to the query parameter, but it also _has_ to be done
+		// specifically for the case of accessing the AMP Customizer (in which customize.php is requested with the query
+		// parameter) in order to prevent the registration of Customizer controls from suppressed plugins. Suppression
+		// could be done early for Transitional mode as well since a query parameter is also used for frontend requests
+		// but there is no similar need to suppress the registration of Customizer controls in Transitional mode since
+		// there is no separate Customizer for AMP in Transitional mode (or legacy Reader mode).
+		if ( $this->is_reader_theme_request() ) {
+			add_action( 'plugins_loaded', [ $this, 'suppress_plugins' ], $priority );
+		} else {
+			// In Standard mode we _have_ to wait for the wp action because with the absence of a query parameter
+			// we have to rely on is_amp_endpoint() and the WP_Query to determine whether a plugin should be suppressed.
+			add_action( 'wp', [ $this, 'maybe_suppress_plugins' ], $priority );
+		}
+	}
+
+	/**
+	 * Is reader theme request.
+	 *
+	 * @return bool
+	 */
+	public function is_reader_theme_request() {
+		return (
+			AMP_Theme_Support::READER_MODE_SLUG === AMP_Options_Manager::get_option( Option::THEME_SUPPORT )
+			&&
+			AMP_Reader_Themes::DEFAULT_READER_THEME !== AMP_Options_Manager::get_option( Option::READER_THEME )
```

Comment:
```suggestion
			! amp_is_legacy()
```

Reply:
I don't think that will work because then it will match `standard` and `transitional` modes as well.

Reply:
Hmm yea that makes sense.